### DISCUSSION TITLE
Masking out cloud values with rioxarray

### DISCUSSION THREAD

**Discussion by nkorinek** [corteva/rioxarray#232] [Q&A] [answered]
Hello!

I am currently trying to figure out the best way to mask out cloud values in a rioxarray object. The Earth Lab team and I have been updating our lessons to lean more on rioxarray then rasterio, and I got a little stuck on this lesson here about masking cloud values: https://www.earthdatascience.org/courses/use-data-open-source-python/multispectral-remote-sensing/landsat-in-Python/remove-clouds-from-landsat-data/. 

As you can see in that lesson, currently we do it by just getting the numpy arrays out of the rioxarray objects with `.values` and then applying a mask to it that with numpy. I'll post the relevant code below, where `landsat_pre` is the landsat image with the clouds, and `landsat_qa` is the layer we used to produce our mask. 

```
landsat_pre_cl_free = em.mask_pixels(
    landsat_pre.values, landsat_qa.values, vals=all_masked_values)
```

`em.mask_pixels` turns the `landsat_qa` array into a mask based on the values in `all_masked_values` and applies it to `landsat_pre`. Sorry if this is confusing! Just want to make sure all the relevant information is here.

The question boils down to this: is there a way to apply a mask like this to a rioxarray object that doesn't involve turning it into a numpy array? Any insight here would be greatly appreciated, thanks for your time!

**Comment by nkorinek** [comment]
Just to maybe clarify a bit more, here's what `em.mask_pixels` is essentially doing behind the scenes:

```
from numpy import ma
all_masked_values = [96, 112, 160, 176, 224]
mask = landsat_qa.isin(all_masked_values)
return ma.masked_array(landsat_pre.values, mask)
```

**Comment by snowman2** [comment, marked as answer]
Looks like xarray also has the [isin](http://xarray.pydata.org/en/stable/generated/xarray.DataArray.isin.html) method.

Side note: xarray doesn't support masked arrays, it just converts to float and uses `nan` [ref](https://stackoverflow.com/questions/41505699/xarray-with-masked-arrays-while-preserving-integer-dtypes).

So, the equivalent xarray form would be:
```python
all_masked_values = [96, 112, 160, 176, 224]
masked = landsat_pre.where(~landsat_qa.isin(all_masked_values))
```

**Comment by nkorinek** [comment]
Thanks for the feedback, that looks great!
